--- quake-1.06/QW/client/gl_vidlinuxglx.c.orig	Tue Feb 25 23:27:22 2003
+++ quake-1.06/QW/client/gl_vidlinuxglx.c	Tue Feb 25 23:31:09 2003
@@ -231,12 +231,21 @@
 
 static void install_grabs(void)
 {
+	Pixmap blank;
+	XColor dummy;
+	char data[1] = {0};
+	Cursor cursor;
+	blank = XCreateBitmapFromData (dpy, win, data, 1, 1);
+	if(blank == None) fprintf(stderr, "error: out of memory.\n");
+	cursor = XCreatePixmapCursor(dpy, blank, blank, &dummy, &dummy, 0, 0);
+	XFreePixmap (dpy, blank);
+	Con_Printf("Cursor blanked.\n");
 	XGrabPointer(dpy, win,
 				 True,
 				 0,
 				 GrabModeAsync, GrabModeAsync,
 				 win,
-				 None,
+				 cursor,
 				 CurrentTime);
 
 #ifdef USE_DGA
