--- quake-1.06/QW/Makefile.Linux.orig	Thu Feb 27 15:49:24 2003
+++ quake-1.06/QW/Makefile.Linux	Thu Feb 27 16:04:38 2003
@@ -14,7 +14,7 @@
 VERSION=2.30$(GLIBC)
 
 ifneq (,$(findstring libc6,$(shell if [ -e /lib/libc.so.6* ];then echo libc6;fi)))
-GLIBC=-glibc
+GLIBC=.glibc
 else
 GLIBC=
 endif
@@ -27,15 +27,13 @@
 RPMARCH=i386
 endif
 
-MAINDIR=/grog/Projects/QW
+MAINDIR=./
 
 BUILD_DEBUG_DIR=debug$(ARCH)$(GLIBC)
 BUILD_RELEASE_DIR=release$(ARCH)$(GLIBC)
 CLIENT_DIR=$(MAINDIR)/client
 SERVER_DIR=$(MAINDIR)/server
 
-MESA_DIR=/usr/local/src/Mesa-3.0
-
 CC=gcc
 BASE_CFLAGS=-Wall -Dstricmp=strcasecmp -I$(CLIENT_DIR) -I$(SERVER_DIR)
 DEBUG_CFLAGS=$(BASE_CFLAGS) -g
@@ -47,13 +45,13 @@
 	-fomit-frame-pointer -fexpensive-optimizations -malign-loops=2 \
 	-malign-jumps=2 -malign-functions=2
 endif
-GLCFLAGS=-DGLQUAKE -I/usr/local/src/Mesa-3.0/include -I/usr/include/glide
+GLCFLAGS=-DGLQUAKE -I/usr/include/glide -I/usr/X11R6/include
 
 LDFLAGS=-lm
 SVGALDFLAGS=-lvga
 XLDFLAGS=-L/usr/X11R6/lib -lX11 -lXext
-GL_SVGA_LDFLAGS=-L/usr/X11/lib -L/usr/local/src/Mesa-3.0/lib -lm -lMesaGL -lglide2x -lX11 -lXext -lvga
-GL_X11_LDFLAGS=-L/usr/X11/lib -L/usr/local/src/Mesa-3.0/lib -lm -lGL -lX11 -lXext
+GL_SVGA_LDFLAGS=-L/usr/X11R6/lib -lm -lglide2x -lX11 -lXext -lvga
+GL_X11_LDFLAGS=-L/usr/X11R6/lib -L/usr/X11R6/include -lm -lGL -lX11 -lXext
 
 DO_CC=$(CC) $(CFLAGS) -o $@ -c $<
 DO_O_CC=$(CC) -O $(CFLAGS) -o $@ -c $<
@@ -71,7 +69,7 @@
 TARGETS=$(BUILDDIR)/qwsv 
 #$(BUILDDIR)/qwcl.x11
 else
-TARGETS=$(BUILDDIR)/qwsv $(BUILDDIR)/qwcl $(BUILDDIR)/qwcl.x11 $(BUILDDIR)/glqwcl $(BUILDDIR)/glqwcl.glx
+TARGETS=$(BUILDDIR)/qwsv $(BUILDDIR)/qwcl $(BUILDDIR)/qwcl.x11 $(BUILDDIR)/glqwcl.glx
 endif
 
 build_debug:
