--- quake-1.06/WinQuake/common.c.orig	Tue Mar  4 13:57:34 2003
+++ quake-1.06/WinQuake/common.c	Tue Mar  4 13:57:36 2003
@@ -1743,7 +1743,7 @@
 	if (i && i < com_argc-1)
 		strcpy (basedir, com_argv[i+1]);
 	else
-		strcpy (basedir, host_parms.basedir);
+		strcpy (basedir, "/usr/share/quake");
 
 	j = strlen (basedir);
 
--- quake-1.06/QW/client/common.c.orig	Tue Mar  4 14:02:28 2003
+++ quake-1.06/QW/client/common.c	Tue Mar  4 14:02:32 2003
@@ -1838,7 +1838,7 @@
 	if (i && i < com_argc-1)
 		strcpy (com_basedir, com_argv[i+1]);
 	else
-		strcpy (com_basedir, host_parms.basedir);
+		strcpy (com_basedir, "/usr/share/quake");
 
 //
 // start up with id1 by default
